libraries/botbuilder-dialogs-adaptive/src/entityAssignments.ts

```diff
@@ -11,28 +11,31 @@ import { EntityAssignment } from './entityAssignment';
 
 const events = 'this.events';
 
+export interface EntityAssignmentsConfiguration {
+    assignments: EntityAssignment[];
+}
+
 /**
  * Tracks entity related events to surface.
  * @remarks When processing entities possible ambiguities are identified and when resolved they turn into assign events.
  * This tracking persists across multiple input utterances.
  */
-export class EntityAssignments {
-    /**
-     * Queue of entity assignments.
-     */
-    public assignments: Partial<EntityAssignment>[] = [];
+export class EntityAssignments implements EntityAssignmentsConfiguration {
+    constructor(public assignments: EntityAssignment[] = []) {}
```

Comment:
Can this be made `readonly`?

Reply:
Negative. It gets modified outside of the class. Does so in .NET, too.